src/tokenizer/shell.cpp

```diff
@@ -0,0 +1,526 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include "tokenizer/shell.hpp"
+#include "utils.hpp"
+
+using namespace std::literals;
+
+namespace ddwaf {
+
+namespace {
+re2::RE2 redirection_regex(
+    R"((&>>?|(?:[0-9]?>(?:\||>|&[0-9]?-?)?)|(?:[0-9]?<(?:<(?:-|<)?|&[0-9]?-?|>)?)))");
+
+bool is_var_char(char c) { return ddwaf::isalnum(c) || c == '_'; }
+
+bool is_field_char(char c)
```

Comment:
Roughly yes, although mostly because some other characters might have a special meaning (or not), so this is just an approximation. I'll add a comment.